### PR TITLE
Added working draft of update notification template code

### PR DESCRIPTION
**Pull request by czpython**
Defaults to `patch` check.
Only shows if user is superuser

image

left to do:
- [x] use correct url for latest.json
- [x] finalise message
- [x] adapt readme

### CODE REVIEW

**Comment by czpython** [comment]
@vxsx You can use this endpoint as a test if needed https://www.dropbox.com/s/4dsnb1ruh0snpp3/latest.json?dl=1

Do you think we should add a <meta> tag with this in it as well?

File: djangocms_admin_style/static/djangocms_admin_style/js/modules/update-notification.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    });
+}
+
+/**
+ * @function cmpVersion
+ * @param {String} a
+ * @param {String} b
+ * @returns {Number}
+ */
+function cmpVersion(a, b) {
```
Comment (FinalAngel) [inline djangocms_admin_style/static/djangocms_admin_style/js/modules/update-notification.js]: ``compareVersion``

File: djangocms_admin_style/static/djangocms_admin_style/js/modules/update-notification.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    return a.length - b.length;
+}
+
+/**
+ * @function gtVersion
+ * @param {String} a
+ * @param {String} b
+ * @returns {Boolean} true if a > b or a === b but a is a dev/rc version
+ */
+function gtVersion(a, b) {
```
Comment (FinalAngel) [inline djangocms_admin_style/static/djangocms_admin_style/js/modules/update-notification.js]: ``greaterThenVersion``

File: djangocms_admin_style/templates/admin/inc/cms_upgrade_notification.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{% load i18n %}
+<meta name="djangocms_version" content="{{ cms_version }}">
+<meta name="djangocms_version_check_type" content="{{ cms_version_check_type }}">
+<script type="text/template" id="cms-update-notification">
+    <div class="cms-update-message" style="display: none;">
```
Comment (FinalAngel) [inline djangocms_admin_style/templates/admin/inc/cms_upgrade_notification.html]: use hidden?

File: djangocms_admin_style/templatetags/admin_style_tags.py [hunk 4]
```diff
@@ -3,7 +3,15 @@
...
 from django import template
+from django.conf import settings
 from django.contrib.sites.shortcuts import get_current_site
+from django.template.loader import render_to_string
+
+
+# We follow the Semantic versioning convention
+# minor - Refers to the minor release track (3.x.1)
+# patch - Refers to the patch release track (3.4.x)
+VALID_VERSION_CHECK_TYPES = ('minor', 'patch')
```
Comment (FinalAngel) [inline djangocms_admin_style/templatetags/admin_style_tags.py]: @czpython we also need to add a setting to disable the check

File: djangocms_admin_style/static/djangocms_admin_style/js/modules/update-notification.js [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+        e.preventDefault();
+
+        Cookies.set(
+            'cms_upgrade_notification_closed',
+            JSON.stringify({
+                version: versionObject.version,
+                type: checkType
+            }),
+            {
+                exprires: MAIN_COOKIE_EXPIRATION
```
Comment (FinalAngel) [inline djangocms_admin_style/static/djangocms_admin_style/js/modules/update-notification.js]: typo

**Comment by FinalAngel** [comment]
Need to adapt the README

**Comment by lory87** [comment]
lgtm

**Comment by FinalAngel** [comment]
@czpython ready to go :)

File: djangocms_admin_style/templates/admin/inc/cms_upgrade_notification.html [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{% load i18n %}
+<meta name="djangocms_version" content="{{ cms_version }}">
+<meta name="djangocms_version_check_type" content="{{ cms_version_check_type }}">
+<script type="text/template" id="cms-update-notification">
+    <div class="cms-update-message" hidden>
+        {% autoescape off %}
+            {% blocktrans with version='<strong>'|add:cms_version|add:'</strong>' latest='<strong class="js-latest-version"></strong>' release_link_before='<a class="js-release-notes-link" target="_blank" href="">' release_link_after='</a>' %}
+                There is a django CMS upgrade available. You are using: {{ version }}, new version is: {{ latest }}.
```
Comment (evildmp) [inline djangocms_admin_style/templates/admin/inc/cms_upgrade_notification.html]: Looks good to me @FinalAngel

File: README.rst [hunk 7]
```diff
@@ -27,6 +27,16 @@ Look at the source of ``templates/admin/base_site.html`` and override the templa
 For example, you can add your own CSS in ``templates/admin/inc/extrastyle.html``.
 
 
+Settings
+========
+
+* ``CMS_ENABLE_UPDATE_CHECK = True``
+  Set to ``False`` to disable the update notification.
+* ``CMS_UPDATE_CHECK_TYPE = ('patch')``
```
Comment (czpython) [inline README.rst]: This is not right.
Can't be set to both, it's either `patch` or `minor`.
Currently it defaults to `minor`.

Comment (FinalAngel) [inline README.rst]: @czpython alright will fix this :)

**Comment by vxsx** [comment]
is this going to be released with 3.4 ? current json shows 3.4 as latest released

**Comment by FinalAngel** [comment]
aye

**Comment by FinalAngel** [comment]
@vxsx what is the status here? ?_?

**Comment by FinalAngel** [comment]
LGTM

**Pull request merged**